examples/bmi-for-nextgen.ipynb

```diff
@@ -0,0 +1,1696 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7b8b986e",
+   "metadata": {},
+   "source": [
+    "# 2 The right way\n",
+    "\n",
+    "Fortunately, we can remedy most of these issues by first making `snowPy` BMI compliant and then implementing the full set of BMI functions.\n",
+    "\n",
+    "## 2.1 BMI background\n",
+    "\n",
+    "The Community Surface Dynamics Modeling System ([CSDMS](https://csdms.colorado.edu/)) group at the University of Colorado Boulder develops and maintains BMI. You can find further information on the [BMI GitHub page](https://github.com/csdms/bmi) or on the [CSDMS Wiki](https://csdms.colorado.edu/wiki/BMI).\n",
+    "\n",
+    "The two reference papers for BMI are:\n",
+    "\n",
+    "- *Hutton, E.W., Piper, M.D., and Tucker, G.E., 2020. The Basic Model Interface 2.0: A standard interface for coupling numerical models in the geosciences. Journal of Open Source Software, 5(51), 2317, https://doi.org/10.21105/joss.02317.*\n",
+    "- *Peckham, S.D., Hutton, E.W., and Norris, B., 2013. A component-based approach to integrated modeling in the geosciences: The design of CSDMS. Computers & Geosciences, 53, pp.3-12, http://dx.doi.org/10.1016/j.cageo.2012.04.002.*\n",
+    "\n",
+    "The CSDMS `heat` [example](https://github.com/csdms/bmi-example-python) is a great reference for BMI implementations. We based the `snowBMI` model, which we'll see shortly, heavily on the CSDMS examples. \n",
+    "\n",
+    "## 2.2 BMI philosophy\n",
+    "\n",
+    "BMI is middleware, meaning it is software that operates between other components (in our case a centralized framework—NextGen—and a model). It standardizes model querying, control, and coupling. \n",
+    "\n",
+    "BMI works because the functions are the same across models and languages. Making a model BMI compliant can remove (or mask) model idiosyncrasies.\n",
+    "\n",
+    "From the [BMI wiki](https://bmi.readthedocs.io/en/stable/): \n",
+    "\n",
+    "> We believe that numerical models, and the sub-components that make up those models, should offer…standardization. To this end, the Community Surface Dynamics Modeling System (CSDMS) has developed the Basic Model Interface (BMI): a set of **standard control and query functions** that, when added to a model code, make that model both easier to learn and easier to couple with other software elements.\n",
+    "\n",
+    "And here's a brief, handy guide to what BMI does and does not do.\n",
+    "\n",
+    "| BMI does… | BMI does not… |\n",
+    "| :-  | :-  |\n",
+    "| Control model runtime using standardized functions | Affect hydrologic model code|\n",
+    "| Pass data in/out of models using standardized functions | Perform unit conversions |\n",
+    "| Provide model and variable information | Perform spatial transforms |\n",
+    "| Work across multiple languages and models | Provide information it’s not asked for |\n",
```

Comment:
Is this from the BMI wiki?  If so, I would say that it is misleading at best.

Reply:
The "Work across multiple languages and models" part, that is...

Reply:
Is "Implemented in different languages" more correcter?

Reply:
I updated the text and added an asterisk for clarification